cpp_faster_fifo/tests/comparison_tests.py

```diff
@@ -0,0 +1,173 @@
+import multiprocessing
+from unittest import TestCase
+from faster_fifo import Queue
+from time import time
+from queue import Full, Empty
+import logging
+
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+
+log = logging.getLogger('rl')
+log.setLevel(logging.DEBUG)
+log.handlers = []  # No duplicated handlers
+log.propagate = False  # workaround for duplicated logs in ipython
+log.addHandler(ch)
+
+MSG_SIZE = 5
+
+
+def make_msg(msg_idx):
+    return (msg_idx,) * MSG_SIZE
+
+
+def produce_msgs(q, p_idx, num_messages):
+    i = 0
+    while i < num_messages:
+        try:
+            q.put(make_msg(i), timeout=0.01)
+            if i % 100000 == 0:
+                log.info('Produce: %d %d', i, p_idx)
+            i += 1
+        except Full:
+            # time.sleep(0.001)
+            pass
+        except Exception as exc:
+            log.exception(exc)
+    log.info('Done producing! %d', p_idx)
+
+
+def consume_msgs(q, queue_type, p_idx, consume_many=None, total_num_messages=int(1e9)):
```

Comment:
I don't think total_num_messages has any meaning in this function. You should remove it I guess. You don't even pass any value for this parameter

Reply:
Done